django/api/services/calculate_rebate.py

```diff
@@ -0,0 +1,70 @@
+from ..models.go_electric_rebate_application import GoElectricRebateApplication
+from ..models.household_member import HouseholdMember
+from ..settings import INCOME_REBATES
+
+
+def calculate_rebate_amount(cra_response, application_id):
+    def check_individual(primary_income):
+        if primary_income is None:
+            return "not approved"
+        primary_income = int(primary_income)
+        if primary_income > INCOME_REBATES.get("C").get("individual_income"):
+            return "not approved"
+        elif primary_income <= INCOME_REBATES.get("A").get("individual_income"):
+            return "A"
+        elif primary_income <= INCOME_REBATES.get("B").get("individual_income"):
+            return "B"
+        elif primary_income <= INCOME_REBATES.get("C").get("individual_income"):
+            return "C"
+
+    def check_household(primary_income, secondary_income):
+        if (primary_income is None) | (secondary_income is None):
+            return "not approved"
+        household_income = int(primary_income) + int(secondary_income)
+        if household_income > INCOME_REBATES.get("C").get("household_income"):
+            return "not approved"
+        if household_income <= INCOME_REBATES.get("A").get("household_income"):
+            return "A"
+        elif household_income <= INCOME_REBATES.get("B").get("household_income"):
+            return "B"
+        elif household_income <= INCOME_REBATES.get("C").get("household_income"):
+            return "C"
+
+    def get_final_rebate(individual_rebate, household_rebate):
+        if household_rebate == "A":
+            return INCOME_REBATES.get("A").get("rebate")
+        if individual_rebate == "B" or household_rebate == "B":
+            return INCOME_REBATES.get("B").get("rebate")
+        if individual_rebate == "C" or household_rebate == "C":
+            return INCOME_REBATES.get("C").get("rebate")
+        if household_rebate == "not approved" and individual_rebate == "not approved":
+            return "not approved"
+
+    application = cra_response.get(application_id)
+    primary_applicant = {}
+    secondary_applicant = {}
+    filtered_applications = GoElectricRebateApplication.objects.filter(
+        id=application_id
+    )
+    filtered_household = HouseholdMember.objects.filter(application=application_id)
+    for idx, x in enumerate(application):
```

Comment:
Use the one to one relationship on an application object to find the house hold member.

Reply:
done